great_expectations/execution_engine/pandas_execution_engine.py

```diff
@@ -340,7 +340,7 @@ def _get_reader_fn(self, reader_method=None, path=None):
                 reader_fn = partial(reader_fn, **reader_options)
             return reader_fn
         except AttributeError:
-            raise ge_exceptions.BatchSpecError(
+            raise ge_exceptions.ExecutionEngineError(
                 f'Unable to find reader_method "{reader_method}" in pandas.'
```

Comment:
These errors were mis-named. The errors come from the ExecutionEngine not being able to resolve reader method from parameters or path, rather than a BatchSpec being instantiated incorrectly